src/js/components/CheckBox/CheckBox.js

```diff
@@ -143,7 +143,9 @@ class CheckBox extends Component {
         as={Box}
         align="center"
         justify="center"
-        margin={{ [side]: theme.checkBox.gap || 'small' }}
+        margin={
+          label && label !== '' && { [side]: theme.checkBox.gap || 'small' }
```

Comment:
I don't think we need the `!== ''` as the prior check for just `label` will be false in this case. Actually though, can we just check whether `label !== undefined`?